torch/_tensor.py

```diff
@@ -1012,6 +1012,34 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
             ret = func(*args, **kwargs)
             return _convert(ret, cls)
 
+    def __dlpack__(self, stream=None):
+        """
+        Creates a dlpack capsule of the current tensor to
+        be exported to other libraries.
+
+        This function will be called from the `from_dlpack` method
+        of the consumer library.
+
+        Args:
+            stream (integer or None): Object that represents a stream and provides
+                a `synchronize` method. Optional.
+        """
+        if isinstance(stream, torch.cuda.Stream) or hasattr(stream, 'synchronize'):
+            stream.synchronize()
+        elif stream is not None and type(stream) is int:
```

Comment:
It would be good to special-case `-1`, since it's the sentinel value for saying explicitly "do not synchronize".